team/bundles/org.eclipse.core.pki/src/org/eclipse/pki/auth/AuthenticationPlugin.java

```diff
@@ -0,0 +1,1133 @@
+
+import org.eclipse.pki.dialog.PassphraseDialog;
+import org.eclipse.pki.exception.UserCanceledException;
+import org.eclipse.pki.jsse.CdeX509TrustManager;
+import org.eclipse.pki.pkcs.VendorImplementation;
+import org.eclipse.pki.pkiselection.PKCSSelected;
+import org.eclipse.pki.pkiselection.PKCSpick;
+import org.eclipse.pki.pkiselection.PKIProperties;
+import org.eclipse.pki.preferences.AuthenticationPreferences;
+import org.eclipse.pki.util.KeyStoreFormat;
+import org.eclipse.pki.util.KeyStoreUtil;
+import org.eclipse.pki.util.LogUtil;
+import org.eclipse.pki.util.PKIAuthenticator;
+import org.eclipse.pki.util.SecurityFileSnapshot;
+import org.eclipse.pki.wizard.PKILoginWizard;
+import org.eclipse.pki.wizard.TrustStoreLoginWizard;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.preferences.ConfigurationScope;
+
+/**
+ * The activator class controls the plug-in life cycle
+ */
+public class AuthenticationPlugin extends AbstractUIPlugin {
+
+    // The plug-in ID
+	
+	private KeyStore userKeyStore = null;
+	private transient String certPassPhrase;
+	private KeyStore trustStore;
+	private transient String trustStorePassPhrase;
+	PKIProperties snapshotProperties = null;
+	private Provider provider;
+	
+	//
+	// Cert defines
+	//
+	private static final int DIGITAL_SIGNATURE = 0;
+	private static final int KEY_CERT_SIGN = 5;
+	private static final int CRL_SIGN = 6;
+	
+	private static final TrustManager trustManager = new CdeX509TrustManager();
+	
+    // The shared instance
+    private static AuthenticationPlugin plugin;
+    
+    protected static final String DEFAULT_TRUST_STORE = "cacerts";
+    protected static final String CONFIGURATION_DIR = "configuration";
+    
+    private static final String DEFAULT_TRUST_STORE_PASSWORD = "changeit";
+    private static final String JAVA_SSL_CFG_PATH_KEY = "java.cfg.file";
+    private static final String JAVA_SSL_TRUST_STORE_PATH_KEY = "javax.net.ssl.trustStore";
+    private static final String JAVA_SSL_TRUST_STORE_TYPE_KEY = "javax.net.ssl.trustStoreType";
+    private static final String JAVA_SSL_TRUST_STORE_PASS_KEY = "javax.net.ssl.trustStorePassword";
+    private static final String JAVA_SSL_USER_KEY_STORE_PATH_KEY = "javax.net.ssl.keyStore";
+    private static final String JAVA_SSL_USER_KEY_STORE_TYPE_KEY = "javax.net.ssl.keyStoreType";
+    private static final String JAVA_SSL_USER_KEY_STORE_PASS_KEY = "javax.net.ssl.keyStorePassword";
+    private static final String JAVA_SSL_USER_KEY_STORE_PROVIDER_KEY = "javax.net.ssl.keyStoreProvider";
+    
+    /**
+     * The constructor
+     */
+    public AuthenticationPlugin() {}
+    
+    /**
+     * Returns the shared instance
+     *
+     * @return the shared instance
+     */
+    public static AuthenticationPlugin getDefault() {
+    	System.out.println("AuthenticationPlugin ---getDefault plugin");
+    	if (plugin == null) {
+    		AuthenticationPlugin auth = new AuthenticationPlugin();

```

Comment:
Do you plan to support running without Platform for your plug-in? In this case I would consider another functionality entry point instead. 
Moreover, activator should be an internal type that implements hooks for Bundle lifecycle. Ideally activator should not be present at all to avoid startup impact.

Reply:
Very good question..  I think the answer should be YES.  Support with or without GUI.  The gotcha on this is
having all the required setup, more so for PKCS11 implementation..